File: test-automation-core/src/main/java/com/nortal/test/core/rest/api/ReportUploaderApi.java
package com.nortal.test.core.rest.api;

import okhttp3.MultipartBody;
import retrofit2.Call;
import retrofit2.http.Headers;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Path;

public interface ReportUploaderApi {

	@Multipart
	@Headers("Content-Encoding: gzip")
	@POST("/v2/testsupport/automation/reports/{buildName}/{buildNumber}")
	Call<Void> upload(@Path("buildName") String buildName, @Path("buildNumber") String buildNumber, @Part MultipartBody.Part filePart);

}

File: test-automation-core-old/src/main/java/com/nortal/test/core/rest/api/ReportUploaderApi.kt
package com.nortal.test.core.rest.api

import okhttp3.MultipartBody
import retrofit2.Call
import retrofit2.http.*
import java.lang.Void

interface ReportUploaderApi {
    @Multipart
    @Headers("Content-Encoding: gzip")
    @POST("/v2/testsupport/automation/reports/{buildName}/{buildNumber}")
    fun upload(@Path("buildName") buildName: String?, @Path("buildNumber") buildNumber: String?, @Part filePart: MultipartBody.Part?): Call<Void?>?
}